Composer/packages/client/src/components/CreationFlow/v2/DefineConversation.tsx

```diff
@@ -0,0 +1,369 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//TODO: Remove Path module
+import Path from 'path';
+
+import { DialogFooter } from 'office-ui-fabric-react/lib/Dialog';
+import formatMessage from 'format-message';
+import { PrimaryButton, DefaultButton } from 'office-ui-fabric-react/lib/Button';
+import { Stack, StackItem } from 'office-ui-fabric-react/lib/Stack';
+import React, { Fragment, useEffect, useCallback, useMemo, useState } from 'react';
+import { TextField } from 'office-ui-fabric-react/lib/TextField';
+import { RouteComponentProps } from '@reach/router';
+import querystring from 'query-string';
+import { FontWeights } from '@uifabric/styling';
+import { DialogWrapper, DialogTypes } from '@bfc/ui-shared';
+import { useRecoilValue } from 'recoil';
+import { Dropdown } from 'office-ui-fabric-react/lib/Dropdown';
+
+import {
+  DialogCreationCopy,
+  QnABotTemplateId,
+  nameRegexV2,
+  invalidNameCharRegex,
+  mockLanguageOptions,
+  runtimeOptions,
+  defaultPrimaryLanguage,
+  defaultRuntime,
+} from '../../../constants';
+import { FieldConfig, useForm } from '../../../hooks/useForm';
+import { StorageFolder } from '../../../recoilModel/types';
+import { createNotification } from '../../../recoilModel/dispatchers/notification';
+import { ImportSuccessNotificationWrapper } from '../../ImportModal/ImportSuccessNotification';
+import { dispatcherState } from '../../../recoilModel';
+import { LocationSelectContent } from '../LocationSelectContent';
+
+// -------------------- Styles -------------------- //
+
+const textFieldlabel = {
+  label: {
+    root: [
+      {
+        fontWeight: FontWeights.semibold,
+      },
+    ],
+  },
+};
+
+const name = {
+  subComponentStyles: textFieldlabel,
+};
+
+const description = {
+  subComponentStyles: textFieldlabel,
+};
+
+const halfstack = {
+  root: [
+    {
+      flexBasis: '50%',
+    },
+  ],
+};
+
+const stackinput = {
+  root: [
+    {
+      marginBottom: '1rem',
+    },
+  ],
+};
+
+// -------------------- DefineConversation -------------------- //
+
+const MAXTRYTIMES = 10000;
+
+interface DefineConversationFormData {
+  name: string;
+  description: string;
+  schemaUrl: string;
+  primaryLanguage: string;
+  runtimeChoice: string;
+  location?: string;
+
+  templateDir?: string; // location of the imported template
+  eTag?: string; // e tag used for content sync between composer and imported bot content
+  urlSuffix?: string; // url to deep link to after creation
+  alias?: string; // identifier that is used to track bots between imports
+  preserveRoot?: boolean; // identifier that is used to determine ay project file renames upon creation
+}
+
+interface DefineConversationProps
```

Comment:
our convention is to use types over interfaces

Reply:
Handled